assets/app/view/game/buy_trains.rb

```diff
@@ -154,6 +156,38 @@ def from_depot(depot_trains)
         end
       end
 
+      def from_depot_discounted(depot_trains, ability)
+        depot_trains.flat_map do |train|
+          next unless ability.trains.include? train.name
+
+          train.variants.sort_by { |_, v| v[:price] }.flat_map do |name, variant|
+            price = variant[:price]
+            price -= if ability.absolute
+                       ability.discount
+                     else
+                       price * (ability.discount / 100.0)
+                     end
+
+            buy_train = lambda do
+              # FIXME: Does not seem to remove company from corporation
+              ability.use!
```

Comment:
you cannot make any changes in the view and you should not

this should all be done in the engine, this is why it does not work

Reply:
OK. Have now pushed up a new commit which does the update in the engine. Now it works.